hollow/src/main/java/com/netflix/hollow/api/consumer/metrics/AbstractRefreshMetricsListener.java

```diff
@@ -0,0 +1,178 @@
+package com.netflix.hollow.api.consumer.metrics;
+
+import com.netflix.hollow.api.consumer.HollowConsumer;
+import com.netflix.hollow.api.custom.HollowAPI;
+import com.netflix.hollow.core.read.engine.HollowReadStateEngine;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Objects;
+
+public class AbstractRefreshMetricsListener extends HollowConsumer.AbstractRefreshListener  {
+
+    private long consecutiveSuccesses;
+    private long consecutiveFailures;
+
+    private Instant lastRefreshInstant;
+    private Instant refreshStart;
+
+    private ConsumerRefreshStatus refreshStatus;                    // success or failure
+    private ConsumerRefreshTransitionType refreshTransitionType;    // snapshot or delta transition
+    private ConsumerRefreshLoadType refreshLoadType;                // "init" indicates first successful load on consumer, "subsequent" indicates subsequent load during refresh
+
+    private long refreshLoadCount;
+
+    // Encapsulate the refresh metrics so that more metrics/properties can be added in future wihtout breaking existing clients.
+    public static class ConsumerRefreshMetrics {
+        private long refreshDurationMillis;
+        private ConsumerRefreshStatus refreshStatus;
+        private ConsumerRefreshTransitionType refreshTransitionType;
+        private ConsumerRefreshLoadType refreshLoadType;
+
+        public ConsumerRefreshMetrics(long refreshDurationMillis, ConsumerRefreshStatus refreshStatus, ConsumerRefreshTransitionType refreshTransitionType, ConsumerRefreshLoadType refreshLoadType) {
+            this.refreshDurationMillis = refreshDurationMillis;
+            this.refreshStatus = Objects.requireNonNull(refreshStatus);
+            this.refreshTransitionType = refreshTransitionType;   // Transition type can be unknown for some failed snapshot or delta transition
+            this.refreshLoadType = Objects.requireNonNull(refreshLoadType);
+        }
+
+        public long getRefreshDurationMillis() {
+            return refreshDurationMillis;
+        }
+        public ConsumerRefreshStatus getRefreshStatus() {
+            return refreshStatus;
+        }
+        public ConsumerRefreshTransitionType getRefreshTransitionType() {
+            return refreshTransitionType;
+        }
+        public ConsumerRefreshLoadType getRefreshLoadType() {
+            return refreshLoadType;
+        }
+    }
+
+    public enum ConsumerRefreshTransitionType {
+        Snapshot("snapshot"),       // refresh involved snapshot transition
+        Delta("delta");             // refresh applied a delta transition
+
+        final private String type;
+        ConsumerRefreshTransitionType(String type) {
+            this.type = type;
+        }
+    }
+
+    public enum ConsumerRefreshLoadType {
+        Init("init"),               // consumer's first-time initialized with a snapshot
```

Comment:
Maybe the longer "initial"? In `HollowConsumer` I recently introduced "initial load" nomenclature (by introducing `getInitialLoad()`).

Reply:
Sure!